### PR TITLE
Merge master to readonly-ref

### PR DESCRIPTION
**Pull request by VSadov**
Merge master to readonly-ref

### CODE REVIEW

**Comment by VSadov** [comment]
@OmarTawfik  let me know if you need to merge your change first

**Comment by VSadov** [comment]
@OmarTawfik - I do not think this will be very conflicting with your changes though. There were some tricky parser changes, but I do not think you've changed parser much.

**Comment by VSadov** [comment]
test this please

**Comment by OmarTawfik** [comment]
I'd like to merge the other month-old PR first :(
There were a few test changes that I'm suspecting they'll conflict. If it is not going to be more additional work for you, or if you are not blocked on this, I hope to get mine in today.
Otherwise, you can go ahead and merge. I don't want to be blocking you.

File: src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs [hunk 1]
```diff
@@ -1541,51 +1541,6 @@ public static IEnumerable<string> ParseConditionalCompilationSymbols(string valu
             return defines.AsEnumerable();
         }
 
-        public static bool TryParseLanguageVersion(string value, out LanguageVersion version)
```
Comment (OmarTawfik) [inline src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs]: Where was this refactored to? I cannot find the added lines anywhere.

**Comment by VSadov** [comment]
@OmarTawfik - i have troubles making this work, there are many changes. I may have to start over, so if your thing is ready, feel free to merge

**Comment by OmarTawfik** [comment]
Ok. Sure. Please merge if you think this is going to delay you.

File: src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs (hunk 1 above)
Comment (OmarTawfik) [inline src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs]: Nevermind. It was refactored and it seems like this is a leftover:
https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/LanguageVersion.cs
#Closed

**Pull request closed without merging**